File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.LinkedList;
import java.util.Queue;
import java.util.StringTokenizer;

class Main{
    private static int n, m, v;
    private static int[][] graph;
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer stk = new StringTokenizer(br.readLine());
        n = Integer.parseInt(stk.nextToken());
        m = Integer.parseInt(stk.nextToken());
        v = Integer.parseInt(stk.nextToken());

        graph = new int[n+1][n+1];
        for(int i=0 ; i<m ; i++){
            stk = new StringTokenizer(br.readLine());
            int sour = Integer.parseInt(stk.nextToken()), des = Integer.parseInt(stk.nextToken());
            graph[sour][des] = graph[des][sour] = 1;
        }
        dfs(v, new boolean[n + 1]);
        System.out.println();
        bfs();
        br.close();
    }

    private static void dfs(int sour, boolean[] visit){
        System.out.print(sour+ " ");
        visit[sour] = true;
        for(int i=1; i<=n ; i++){
            if(graph[sour][i] == 0 || visit[i])continue;
            dfs(i, visit);
        }
    }

    private static void bfs(){
        Queue<Integer> que = new LinkedList<>();
        que.add(v);
        boolean[] visit = new boolean[n + 1];
        visit[v] = true;

        while(!que.isEmpty()){
            int sour = que.poll();
            System.out.print(sour + " ");
            for(int i=1 ; i<=n ; i++){
                if(graph[sour][i] == 0 || visit[i])continue;
                que.add(i);
                visit[i] = true;
            }
        }
    }
}

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
from sys import stdin
from collections import deque

n, m, v = map(int, stdin.readline().split())
graph = [[0] * (n + 1) for _ in range(n + 1)]
for _ in range(m):
    sour, des = map(int, stdin.readline().split())
    graph[sour][des] = graph[des][sour] = 1

def bfs(start: int) -> None:
    global graph, n
    visit = [False] * (n + 1)
    visit[start] = True
    que = deque([start])

    print()
    while que:
        sour = que.popleft()
        print(sour, end=' ')
        for des in range(1, n + 1):
            if not visit[des] and graph[sour][des]:
                visit[des] = True
                que.append(des)

def dfs(sour: int) -> None:
    global graph,  n, visit
    if visit[sour]:
        return
    print(sour, end=' ')
    visit[sour] = True
    for des in range(1, n + 1):
        if not visit[des] and graph[sour][des]:
            dfs(des)



visit = [False] * (n + 1)
dfs(v)
bfs(v)
